cls/SourceControl/Git/Utils.cls

```diff
@@ -400,6 +403,13 @@ ClassMethod NewBranch(newBranchName As %String) As %Status
 
 ClassMethod SwitchBranch(targetBranchName As %String) As %Status
 {
+    // Make sure that branch exists first
```

Comment:
What is current behavior (prior to this change) if you pick a branch that already exists on the remote but isn't local yet? What is the behavior after this change?

Reply:
Now both should succeed